wordpress-mobile/gutenberg-mobile
# Point to gutenberg hash with list block spaces crash fix

Fixes #871 

Gutenberg side PR: https://github.com/WordPress/gutenberg/pull/15021

This PR brings in the fix, which is only on source code on the Gutenberg repo.

The internal representation of rich-text in Aztec and the format-lib are not fully compatible or in-sync and we also use html to establish the communication between the two. That can lead to cases like the issue linked, where Aztec will do some trimming to remove spaces that are unimportant for html rendering, but format-lib won't so, making the caret positioning logic to get out of sync.

This fix will try to detect when such spaces will be removed and avoid telling Aztec where to put the caret and hope for the best 🤞.

This fix is not "perfect" though. It only touches the Android side (iOS seems to handle the caret setting OK) and will also lead to the text entered by the user being mutated since the spaces will get trimmed by Aztec.

### To test A

1. Begin editing new list
2. Add **two** spaces, then any letter and then press Enter
3. The list should create a new list item and the spaces in the item above it will be trimmed

### To test B

1. Begin editing new list
2. Add **three** spaces and press Enter
3. The list should create a new list item and the spaces in the item above it will be trimmed

### To test C

1. Begin editing new list
2. Add any letter, add **three** spaces (second space creates a period) and press Enter
3. The list should create a new list item and the spaces in the item above it will be trimmed
